src/Admin/AdminHelper.php

```diff
@@ -98,19 +98,36 @@ public function getAdmin($code)
      * Note:
      *   This code is ugly, but there is no better way of doing it.
      *
-     * @param object $subject
+     * @param object $deprecatedSubject
      * @param string $elementId
      *
      * @throws \RuntimeException
      * @throws \Exception
      *
      * @return array
      */
-    public function appendFormFieldElement(AdminInterface $admin, $subject, $elementId)
+    public function appendFormFieldElement(AdminInterface $admin, $deprecatedSubject, $elementId) // NEXT_MAJOR remove the $subject param
     {
         // child rows marked as toDelete
         $toDelete = [];
+
         // retrieve the subject
+        $subject = $admin->getSubject();
```

Comment:
Shouldn't you check if the admin `hasSubject`?

Reply:
Done @core23